nucypher/utilities/prometheus/collector.py

```diff
@@ -434,63 +433,19 @@ def _event_occurred(self, event) -> None:
-
-        def _event_occurred(self, event) -> None:
-            super()._event_occurred(event)
-            self.metrics[BidRefundCompositeEventMetricsCollector.COMMON_METRIC_KEY].set(
-                self.contract_agent.get_deposited_eth(self.staker_address))
-
-    def __init__(self, staker_address: ChecksumAddress, contract_registry: BaseContractRegistry, metrics_prefix: str):
-        # Bid/Refund (Modify the same metric)
-        worklock_agent = ContractAgency.get_agent(WorkLockAgent, registry=contract_registry)
-
-        self.collectors: List[BidRefundCompositeEventMetricsCollector.BidRefundCommonCollector] = [
-            # Bid Events
-            self.BidRefundCommonCollector(
-                event_name='Bid',
-                event_args_config={
-                    "depositedETH": (Gauge, f'{metrics_prefix}_worklock_bid_depositedETH', 'Deposited ETH value')
-                },
-                argument_filters={"sender": staker_address},
-                staker_address=staker_address,
-                contract_agent=worklock_agent),
-            # Refund Events
-            self.BidRefundCommonCollector(
-                event_name='Refund',
-                event_args_config={
-                    "refundETH": (Gauge, f'{metrics_prefix}_worklock_refund_refundETH', 'Refunded ETH')
-                },
-                argument_filters={"sender": staker_address},
-                staker_address=staker_address,
-                contract_agent=worklock_agent)
-        ]
+    def __init__(self, staker_address: ChecksumAddress, event_name: str = 'Refund', *args, **kwargs):
+        super().__init__(event_name=event_name, argument_filters={'sender': staker_address}, *args, **kwargs)
+        self.staker_address = staker_address
 
     def initialize(self, metrics_prefix: str, registry: CollectorRegistry) -> None:
-        common_gauge = Gauge(f'{metrics_prefix}_worklock_current_deposited_eth',
-                             'Worklock deposited ETH',
-                             registry=registry)
-        for collector in self.collectors:
-            collector.initialize(metrics_prefix, registry)
-            # manually add common gauge to both collectors' dictionary of metrics
-            collector.add_common_metric(common_gauge)
+        super().initialize(metrics_prefix=metrics_prefix, registry=registry)
+        self.metrics["worklock_deposited_eth_gauge"] = Gauge(f'{metrics_prefix}_worklock_current_deposited_eth',
+                                                             'Worklock deposited ETH',
+                                                             registry=registry)
 
-    def collect(self) -> None:
-        for collector in self.collectors:
-            collector.collect()
+    def _event_occurred(self, event) -> None:
+        super()._event_occurred(event)
+        self.metrics["worklock_deposited_eth_gauge"].set(self.contract_agent.get_deposited_eth(self.staker_address))
```

Comment:
Sorry for dumb question: I see name of class includes "refund" and here "deposited", could you explain a little bit?

Reply:
Not dumb at all. Original code from P2O - both Bid and Refund events were tracked and when either event occurred, in addition to event information, they also updated the common metric `worklock_deposited_eth_gauge` - the thought was that bid increased the deposited ETH and refund decreased it. The naming could be improved - but that would involve a change to the dashboard as well.

Reply:
Still do not understand everything, I mean bid is removed, and name contains refund but metrics for deposit. Anyway if it's on purpose then ok

Reply:
I agree this reads a bit confusing - why set `contract_agent.get_deposited_eth` for the *Refund* event?

Reply:
When a refund occurs for WorkLock the following also happens:
```python
info.depositedETH = info.depositedETH.sub(refundETH);
```
And I think this is what is attempted to be captured.